CONTRIBUTING.md

```diff
@@ -1,78 +1,116 @@
-# Contributing Guidelines
-If you're interested in contributing to this project, here are a few ways to do so:
+# Contribution Guidelines
 
-**Note:** If these contributing guidelines are not followed we will close the issue or PR, please
-read these instructions carefully.
+**Note:** If these contribution guidelines are not followed your issue or PR might be closed, so
+please read these instructions carefully.
 
-## Bug fixes
+## Contribution types
+
+### Bug fixes
  - If you find a bug, please first report it using
- [Github issues](https://github.com/flame-engine/flame/issues/new).
+   [Github issues](https://github.com/flame-engine/flame/issues/new).
    - First check if there is not already an issue for it; duplicated issues will be closed.
  - Issues that have already been identified as a bug will be labelled "bug".
- - If you'd like to submit a fix for a bug, send a
- [Pull Request](https://guides.github.com/activities/forking/#making-a-pull-request) from your own
- fork, also read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+ - If you'd like to submit a fix for a bug, please read the [How To](#how-to-contribute) for how to
+   send a Pull Request.
  - Include a test that isolates the bug and verifies that it was fixed.
  - Also update the examples and documentation if necessary.
 
```

Comment:
Also, it would be great to have a short explanation here about what constitutes a great bug report.

Reply:
This document is already so long though, I think that the bug report template should guide most people to write a good bug report and if they still don't, I doubt that they have taken the time to read this document anyways.
But if you have a one-liner or such I could include that?

Reply:
Hmm, so are you in general against this document being very detailed, or merely that we're trying to squeeze too many updates into a single PR?

Reply:
Depends on what `very detailed` entails I think, I would like to have this as short as possible, but I also don't want us to miss out to tell our contributors something crucial. I don't think an example bug report would give anything here, since they already should get that information from the template when they are filling out the bug report, and if there is something missing we could add that to the template. I just know that people generally are quite lazy, and the shorter the doc is, the more likely it is that they will make it through it. :smile:

Reply:
> the shorter the doc is, the more likely it is that they will make it through it. 

Agree with this. 

At the same time, this document already tries to describe multiple separate topics:
 - how to file issues;
 - how to set up development environment;
 - how to create a good PR.

In theory, these could be split into separate documents. (They could even be hosted at the main Documentation site).

Reply:
All of those topic fall under the super-topic that this document is about though; how to contribute.
I think it will be even less likely that people would read what they need if we split it up into more docs, and it would also not follow the GitHub guidelines.